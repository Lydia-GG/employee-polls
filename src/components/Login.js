import React, { useState } from 'react';
import { connect } from 'react-redux';
import { handleLogin } from '../redux/actions/authedUser';
import { useLocation, useNavigate } from 'react-router-dom';
import login from '../assets/login.png';
import Button from './Button';
import Input from './Input';

const Login = (props) => {
  const navigate = useNavigate();
  const location = useLocation();
  const [username, setUsername] = useState('');
  const [password, setPassword] = useState('');

  const handleSubmit = (e) => {
    e.preventDefault();
    props.dispatch(handleLogin(username, password));
    navigate(location.state?.path || '/');
  };

  return (
    <div>
      <h2>Employee Polls</h2>
      <img className="poll-image" src={login} alt="login" />
      <form className="login-form" onSubmit={handleSubmit}>
        <h3>Log In</h3>
        <label>User</label>

        <Input
          type="text"
          placeholder="username"
          className="login-input"
          value={username}
          onChange={(e) => setUsername(e.target.value)}
        />
        <label>Password</label>

        <Input
          type="password"
          placeholder="password"
          className="login-input"
          value={password}
          onChange={(e) => setPassword(e.target.value)}
        />
        <Button
          text="Login"
          onClick={handleSubmit}
          className="login-btn"
          type="submit"
        />
      </form>
      <div className="notes">
        <p>You can use:</p>
        <p>username: tylermcginnis</p>
        <p>password: 123456</p>
      </div>
    </div>
  );
};

const mapStateToProps = ({ authedUser }) => ({
  authedUser,
});
export default connect(mapStateToProps)(Login);
