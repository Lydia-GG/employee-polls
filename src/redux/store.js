import { createStore } from 'redux';
import reducer from './reducers';
import middleware from './middleware';
import { composeWithDevTools } from 'redux-devtools-extension';

const store = createStore(reducer, composeWithDevTools(middleware));

export default store;
