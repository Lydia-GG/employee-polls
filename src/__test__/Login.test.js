/* eslint-disable testing-library/render-result-naming-convention */
/* eslint-disable no-unused-vars */
import { render, screen, fireEvent } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/store';
import { MemoryRouter } from 'react-router-dom';
import Login from '../components/Login';

describe('Login', () => {
  it('expect(component).toMatchSnapshot();', () => {
    const component = render(
      <MemoryRouter>
        <Provider store={store}>
          <Login />
        </Provider>
      </MemoryRouter>
    );
    expect(component).toMatchSnapshot();
  });
  it('inputs should be initially empty', () => {
    render(
      <MemoryRouter>
        <Provider store={store}>
          <Login />
        </Provider>
      </MemoryRouter>
    );

    const userNameInput = screen.getByPlaceholderText('username');
    const passwordInput = screen.getByPlaceholderText('password');

    expect(userNameInput.value).toBe('');
    expect(passwordInput.value).toBe('');
  });
  it('Correctly changes input fields value', () => {
    render(
      <MemoryRouter>
        <Provider store={store}>
          <Login />
        </Provider>
      </MemoryRouter>
    );

    const userNameInput = screen.getByPlaceholderText('username');
    const passwordInput = screen.getByPlaceholderText('password');

    fireEvent.change(userNameInput, { target: { value: 'mary' } });
    fireEvent.change(passwordInput, { target: { value: 123456789 } });

    expect(userNameInput.value).toBe('mary');
    expect(passwordInput.value).toBe('123456789');
  });
});
