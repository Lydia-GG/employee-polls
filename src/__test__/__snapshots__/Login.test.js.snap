// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login expect(component).toMatchSnapshot(); 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h2>
          Employee Polls
        </h2>
        <img
          alt="login"
          class="poll-image"
          src="login.png"
        />
        <form
          class="login-form"
        >
          <h3>
            Log In
          </h3>
          <label>
            User
          </label>
          <input
            class="login-input"
            placeholder="username"
            type="text"
            value=""
          />
          <label>
            Password
          </label>
          <input
            class="login-input"
            placeholder="password"
            type="password"
            value=""
          />
          <button
            class="login-btn"
            type="submit"
          >
            Login
          </button>
        </form>
        <div
          class="notes"
        >
          <p>
            You can use:
          </p>
          <p>
            username: tylermcginnis
          </p>
          <p>
            password: 123456
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h2>
        Employee Polls
      </h2>
      <img
        alt="login"
        class="poll-image"
        src="login.png"
      />
      <form
        class="login-form"
      >
        <h3>
          Log In
        </h3>
        <label>
          User
        </label>
        <input
          class="login-input"
          placeholder="username"
          type="text"
          value=""
        />
        <label>
          Password
        </label>
        <input
          class="login-input"
          placeholder="password"
          type="password"
          value=""
        />
        <button
          class="login-btn"
          type="submit"
        >
          Login
        </button>
      </form>
      <div
        class="notes"
      >
        <p>
          You can use:
        </p>
        <p>
          username: tylermcginnis
        </p>
        <p>
          password: 123456
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
